src/request/actions/ActionCreator.ts

```diff
@@ -0,0 +1,32 @@
+import { Action } from 'redux';
+
+interface IActionCreatorAction<TPayload> extends Action {
+    payload: TPayload;
+}
+
+interface IActionCreatorBase<TPayload> {
+    type: string;
+    unwrap(action: Action): TPayload;
+}
+
+interface IActionCreator<TPayload> extends IActionCreatorBase<TPayload> {
+    (payload: TPayload): IActionCreatorAction<TPayload>;
+}
+
+export function createActionCreator<TPayload>(type: string): IActionCreator<TPayload> {
+    const actionCreator = (payload: TPayload) => {
+        return {
+            type: type,
+            payload: payload
+        }
+    };
+    
+    const typedActionCreator = <IActionCreator<TPayload>>actionCreator;
+    
+    typedActionCreator.type = type;
+    typedActionCreator.unwrap = (action: Action) => {
+        return (<IActionCreatorAction<TPayload>>action).payload;
+    };
```

Comment:
Yes, the `Action` comes into the reducer untyped; it's only after the switch do you know the expected type.  You could arbitrarily cast it to the correct type, but now there's a disconnect between the action type and interface.  The `unwrap()` function is an attempt to couple the two together.  (It also makes the reducer easier to read by eliminating the TypeScript casting, which I find hard to read.)